test/integration/integration_test_fixtures.py

```diff
@@ -46,6 +46,8 @@ def __init__(self, ip_version):
     self.server_ip = "::1" if ip_version == IpVersion.IPV6 else "127.0.0.1"
     self.socket_type = socket.AF_INET6 if ip_version == IpVersion.IPV6 else socket.AF_INET
     self.test_server = None
```

Comment:
Keeping both self.test_server and self.test_servers can be confusing to new readers. Would it be too much work to switch all existing code to using self.test_servers ? (with index of zero as appropriate)

Alternatively, can we add a comment explaining when is self.test_server expected to remain None?